backend/degree/views.py

```diff
@@ -1,3 +1,68 @@
 from django.shortcuts import render
+from rest_framework import generics
+from PennCourses.docs_settings import PcxAutoSchema, reverse_func
+
+from degree.models import (
+    Degree,
+    DegreeRequirement,
+    DegreeFulfillment,
+    DegreePlan
+)
+
+from degree.serializers import (
+    DegreeSerializer,
+    DegreePlanSerializer,
+    DegreeFulfillmentSerializer,
+    DegreeRequirementSerializer
+)
 
 # Create your views here.
+
+
+class DegreeList(generics.ListAPIView):
+    """
+    Retrieve a list of (all) degrees available.
+    """
+
+    schema = PcxAutoSchema(
+        response_codes={
+            reverse_func("degree-list"): {
+                "GET": {200: "[DESCRIBE_RESPONSE_SCHEMA]Courses listed successfully."}
+            }
+        },
+    )
+
+    serializer_class = DegreeSerializer
+
+    # TODO: Actually return a list of possible degrees
+    def get_queryset(self):
+        queryset = Degree.filter()
```

Comment:
Degree.objects.all(); also since you're not including any additional logic here, you can just set the `queryset` attribute directly:
```
queryset = Degree.objects.all()
```